beacon-chain/operations/voluntaryexits/pool.go

```diff
@@ -0,0 +1,167 @@
+package voluntaryexits
+
+import (
+	"math"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	doublylinkedlist "github.com/prysmaticlabs/prysm/v3/container/doubly-linked-list"
+	"github.com/prysmaticlabs/prysm/v3/crypto/bls/blst"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+	"github.com/sirupsen/logrus"
+)
+
+// PoolManager maintains pending and seen voluntary exits.
+// This pool is used by proposers to insert voluntary exits into new blocks.
+type PoolManager interface {
```

Comment:
I was thinking that perhaps we could use generic interfaces for pools, as they all seem
to have the same behavior. For example, they all require fetching
pending items, items for inclusion, insert, and mark as included

```go
type Pool[T any] interface {
	Pending() ([]T, error)
	ForInclusion(state state.ReadOnly, slot types.Slot) ([]T, error)
    Insert(t T)
    MarkIncluded(T)
}
```

What do you think? All our pools I believe could implement this generic
interface which could be much cleaner overall

Reply:
We talked already offline, but just to keep around here, the problem is that the behavior of different pools is quite different. We have different validation routines, expiration/pruning algos, etc. I feel like having generics here will require passing a bunch of enclosures to the exposed functions in the Pool. For example one would need something like 
```
ForInclusion(state state.ReadOly, slot types.Slot, validate func(t T) bool) ([]T, error)
```
Instead of the above.